src/ImageSharp/Formats/Tiff/TiffImageFormatDetector.cs

```diff
@@ -26,9 +34,49 @@ public IImageFormat DetectFormat(ReadOnlySpan<byte> header)
 
         private bool IsSupportedFileFormat(ReadOnlySpan<byte> header)
         {
-            return header.Length >= this.HeaderSize &&
-                   ((header[0] == 0x49 && header[1] == 0x49 && header[2] == 0x2A && header[3] == 0x00) || // Little-endian
-                    (header[0] == 0x4D && header[1] == 0x4D && header[2] == 0x00 && header[3] == 0x2A)); // Big-endian
+            if (header.Length >= this.HeaderSize)
+            {
+                if (header[0] == 0x49 && header[1] == 0x49)
+                {
+                    // Little-endian
+                    if (this.isBigTiff is false)
```

Comment:
That's an odd way to say `!this.isBigTiff`

Reply:
I would actually use a single instance of the detector and create two separate private methods called from here and test against both.

Reply:
done